### DISCUSSION TITLE
Climate Data and Analytics for TraderX

### DISCUSSION THREAD

**Discussion by lucaborella89** [finos/traderX#191] [Ideas]
# Climate Scenario analysis within TraderX 

<img width="814" alt="image (1)" src="https://github.com/user-attachments/assets/689fc84c-610e-4a98-b005-e784d53a019a">

## 07/16/24: FINOS Tech Sprint 2024 Kick-off

1. [Watch the recording](https://www.finos.org/hubfs/FINOS%20events/2024-07%20Tech%20Sprint/Use%20Case%20Pitches/Tech%20Sprint%20Use%20Cases-%20OS-C%20Climate%20Data%20and%20Analytics%20for%20TraderX%20%23191%20CORRECTED.mp4)

2. [A Conceptual Framework for Climate Scenario Analysis in the Trading Book (ISDA 2023)](https://www.isda.org/a/boogE/A-Conceptual-Framework-for-Climate-Scenario-Analysis-in-the-Trading-Book.pdf)

3. [Climate Risk Scenario Analysis for the Trading Book (ISDA 2024)](https://www.isda.org/a/YHlgE/Climate-Scenario-Analysis-in-the-Trading-Book-%E2%80%93-Phase-II.pdf)

4. [Working ppt](https://docs.google.com/presentation/d/1e3oyAjSw6Jyo3p2w7UnGHad0n4B26eQhyUyaoCWu6Y4/edit#slide=id.g2e489ab8cd3_0_0)

## Business Problem
Climate risk analysis has become a regulatory requirement in many countries already. Financial institutions need to integrate climate risk assessments into their trading books. Current frameworks often underestimate the impact of climate-related damages on economic factors. There is a need for a comprehensive tool that can simulate and analyze the effects of physical and transition risks on market and credit risks in trading activities.

## Proposed Solution
Following ISDA recommendation, develop a scenario analysis framework within TraderX, leveraging WITNESS climate and economic scenario data. The scenario analysis framework maps WITNESS output with market risk and credit risk shocks, which are typically used within trading book risk engines. 

## Deliverables
The goal is to develop a proof-of-value (POV) for the business and proof-of-concept (POC) for the technological feasibility that simulates the effects of physical risk models and transition risk models on market risk and credit risk in trading activities, taking as the initial example a securitised portfolio of mortgages (MBS). 

## Roadmap
1. Users simulate various climate and economic scenarios using WITNESS
2. Build a risk engine on top of TraderX 
3. Assess their impacts on the trading book by computing changes in P&L

## High-Level Architecture
TBD

## Related Technologies/Platforms
1. OS-Climate Data Commons
2. Data Dynamics, free data analytics platform for market participants to evaluate and analyze Fannie Mae's vast amount of available data.

## Key people helping to facilitate this hackathon stream:
- Project lead: David Kelly (BIP) 
- Vincent Caldeira (RedHat) 
- Matthieu Meaux (Capgemini)

### Challenge4Pros
Enhancing the WITNESS model by more comprehensively integrating physical risks into its damage functions. While physical risks (like extreme weather events) are already considered in the context of capital (e.g., damage to infrastructure), a more detailed approach is suggested: 
- Distributing Capital: Spread capital geographically based on factors like population density.
- Applying Risks: Apply physical risks to not just capital, but also labor and energy resources within macroeconomic functions.
- Surrogate Models: Use Coupled Model Intercomparison Project Phase 6 (CMIP6) data to create surrogate models for distributing detailed information across the earth, thereby applying physical risk to transition scenarios.

This approach aims to provide a more nuanced and accurate representation of how physical risks impact various economic inputs during climate transitions.

**Comment by lucaborella89** [comment]
@DovOps any suggestions re which risk engine tool could fit well with TraderX? Have you already come across this topic in the past?

**Reply by lucaborella89** [reply]
@johnmark fyi relevant to the synthetic data use case. Any OS credit risk analytics we could use to bridge the gap between TraderX and OS-Climate?

**Comment by lucaborella89** [comment]
27/08 OS-Climate <> Radius <> Backstage meeting notes: 

Example of a POC Team Componentry
1. Morgan Stanley Team - TraderX Application
2. Fannie Mae Team - MBS data sample(s)
3. BIP Team - Mortgage Physical Risk & Pricing Impact
4. Capgemini Team - Transition Risk & Pricing Impact
5. Red Hat Team - Climate Data Mesh
6. Microsoft Azure Team - Cloud Infrastructure

Key Success Factors:
1. Use Case - tight boundary conditions to make it feasible and manageable in scale
2. Example UC = US, Flood Risk, specific States - e.g. Florida, NY, Texas, California
3. End-User Reference/Working Group - need for UX specification, regular show & tells, iterations, refinements, changes, focus, etc

Submitted a data request via OS-C for Fannie Mae data https://github.com/os-climate/Data-Requests/issues/5

Target Dates for a Sample MVP Development
- Sep-2024 = Framework and roadmap (4 weeks)
- Monthly Sprints = 4 week sprints over 6 months
- Deliver business and technology value Monthly

Proposed goals for Sept 30 
P1: Deploy TraderX to Azure, AWS and to on-premise and view those deployments via Backstage dashboard
P2 - Illustrate how an Ops team can use Radius recipes to customize the dependent infrastructure for TraderX
P3 - stretch goal - show CI/CD integration with TraderX

Next steps make sense for all who want to contribute to demo
1) View the Intro to TraderX+Radius [video](https://www.finos.org/hubfs/Tech%20Sprint%202024/Tech%20Sprint%20Use%20Cases-%20TraderX%20deployment%20and%20orchestration%20%23190%20CORRECTED.mp4)
2) Review the PR for deploying TraderX via Radius (Jon will send a link out for this)
3) Followup call for Q&A and discussion
4 )Start defining and assigning specific work items for upcoming 9/12 f:f and the 9/30 demo
5) Setup weekly call for syncing on the above (Keep this aligned with existing office hours call per Karl, every other Friday TraderX call)
6) Business Conversation with MS Front Office and Infra

Other relevant links: https://www.openriskmanagement.com/

**Reply by jonvsm** [reply]
Per above, here is the link for "2. Review the PR for deploying TraderX via Radius (Jon will send a link out for this)" - https://github.com/finos/traderX/pull/204

**Comment by lucaborella89** [comment]
Check this out: 
From @open-risk SFPL Portfolio is a Django app to store Single Family Portfolio Data 
https://github.com/open-risk/openNPL/tree/master/sflp_portfolio

**Comment by lucaborella89** [comment]
Updated ppt here https://docs.google.com/presentation/d/1e3oyAjSw6Jyo3p2w7UnGHad0n4B26eQhyUyaoCWu6Y4/edit

**Comment by lucaborella89** [comment]
https://events.isda.org/workshop-climate
